pkg/generated/openapi/zz_generated.openapi.go

```diff
@@ -49666,6 +49666,14 @@ func schema_k8sio_kube_proxy_config_v1alpha1_KubeProxyIPTablesConfiguration(ref
 							Format:      "",
 						},
 					},
+					"nodePortsOnLocalhost": {
+						SchemaProps: spec.SchemaProps{
+							Description: "nodePortsOnLocalhost tells kube-proxy whether to allow NodePort services can be accessed on localhost.",
+							Default:     false,
```

Comment:
IIRC the flag defaults to true, but configuration override flags

https://github.com/kubernetes/kubernetes/blob/4763d7529097c87f03a73ba9baf48f8ec1b2e1bb/cmd/kube-proxy/app/server.go#L488-L523

so, this will be false by default?
can you verify it?

Reply:
I don't remember if is possible to defaut a bool to true or we had to define it as a pointer
https://github.com/kubernetes/kubernetes/blob/4763d7529097c87f03a73ba9baf48f8ec1b2e1bb/pkg/proxy/apis/config/v1alpha1/defaults.go#L38

Reply:
@aojea  I have verified it, this is true by default. I compiled kube-proxy and ran it, from the log output the flag ` --nodeports-on-localhost` is true by default, please refer to the following:
```bash
cyclinder@CycLinder-4 ~/Desktop/code/kubernetes/_output/bin$ ll kube-proxy                                                                                                       ✹add_flag_in_proxy 
-rwxr-xr-x  1 cyclinder  staff    55M Mar 10 20:38 kube-proxy
cyclinder@CycLinder-4 ~/Desktop/code/kubernetes/_output/bin$ ./kube-proxy -v 1                                                                                                   ✹add_flag_in_proxy 
...
I0310 20:53:04.666943   84609 flags.go:64] FLAG: --master=""
I0310 20:53:04.666947   84609 flags.go:64] FLAG: --metrics-bind-address="127.0.0.1:10249"
I0310 20:53:04.666952   84609 flags.go:64] FLAG: --metrics-port="10249"
I0310 20:53:04.666957   84609 flags.go:64] FLAG: --nodeport-addresses="[]"
I0310 20:53:04.666974   84609 flags.go:64] FLAG: --nodeports-on-localhost="true"
I0310 20:53:04.666979   84609 flags.go:64] FLAG: --one-output="false"
I0310 20:53:04.666983   84609 flags.go:64] FLAG: --oom-score-adj="-999"
I0310 20:53:04.666987   84609 flags.go:64] FLAG: --profiling="false"
I0310 20:53:04.666993   84609 flags.go:64] FLAG: --proxy-mode=""
I0310 20:53:04.667002   84609 flags.go:64] FLAG: --proxy-port-range=""
I0310 20:53:04.667013   84609 flags.go:64] FLAG: --show-hidden-metrics-for-version=""
I0310 20:53:04.667019   84609 flags.go:64] FLAG: --skip-headers="false"
I0310 20:53:04.667263   84609 flags.go:64] FLAG: --skip-log-headers="false"
I0310 20:53:04.667277   84609 flags.go:64] FLAG: --stderrthreshold="2"
I0310 20:53:04.667284   84609 flags.go:64] FLAG: --udp-timeout="250ms"
I0310 20:53:04.667291   84609 flags.go:64] FLAG: --v="1"
I0310 20:53:04.667298   84609 flags.go:64] FLAG: --version="false"
I0310 20:53:04.667308   84609 flags.go:64] FLAG: --vmodule=""
I0310 20:53:04.667316   84609 flags.go:64] FLAG: --write-config-to=""
I0310 20:53:04.667346   84609 server.go:228] "Warning, all flags other than --config, --write-config-to, and --cleanup are deprecated, please begin using a config file ASAP"
I0310 20:53:04.667465   84609 feature_gate.go:245] feature gates: &{map[]}
I0310 20:53:04.667616   84609 feature_gate.go:245] feature gates: &{map[]}
I0310 20:53:04.667744   84609 server.go:563] "Neither kubeconfig file nor master URL was specified, falling back to in-cluster config"
```

when I set `nodeports-on-localhost` = false:
```bash
cyclinder@CycLinder-4 ~/Desktop/code/kubernetes/_output/bin$ ./kube-proxy -v 1 --nodeports-on-localhost=false                                                             130 ↵  ✹add_flag_in_proxy 
I0310 20:55:16.409190   84729 flags.go:64] FLAG: --add-dir-header="false"
I0310 20:55:16.409849   84729 flags.go:64] FLAG: --alsologtostderr="false"
I0310 20:55:16.409858   84729 flags.go:64] FLAG: --bind-address="0.0.0.0"
I0310 20:55:16.409880   84729 flags.go:64] FLAG: --bind-address-hard-fail="false"
I0310 20:55:16.409888   84729 flags.go:64] FLAG: --boot-id-file="/proc/sys/kernel/random/boot_id"
I0310 20:55:16.409896   84729 flags.go:64] FLAG: --cleanup="false"
I0310 20:55:16.409903   84729 flags.go:64] FLAG: --cluster-cidr=""
I0310 20:55:16.409911   84729 flags.go:64] FLAG: --config=""
I0310 20:55:16.409918   84729 flags.go:64] FLAG: --config-sync-period="15m0s"
I0310 20:55:16.409934   84729 flags.go:64] FLAG: --conntrack-max-per-core="32768"
I0310 20:55:16.409943   84729 flags.go:64] FLAG: --conntrack-min="131072"
I0310 20:55:16.409950   84729 flags.go:64] FLAG: --conntrack-tcp-timeout-close-wait="1h0m0s"
I0310 20:55:16.409957   84729 flags.go:64] FLAG: --conntrack-tcp-timeout-established="24h0m0s"
I0310 20:55:16.410281   84729 flags.go:64] FLAG: --detect-local-mode=""
I0310 20:55:16.410294   84729 flags.go:64] FLAG: --feature-gates=""
I0310 20:55:16.410312   84729 flags.go:64] FLAG: --healthz-bind-address="0.0.0.0:10256"
I0310 20:55:16.410322   84729 flags.go:64] FLAG: --healthz-port="10256"
I0310 20:55:16.410329   84729 flags.go:64] FLAG: --help="false"
I0310 20:55:16.410336   84729 flags.go:64] FLAG: --hostname-override=""
I0310 20:55:16.410343   84729 flags.go:64] FLAG: --iptables-masquerade-bit="14"
I0310 20:55:16.410350   84729 flags.go:64] FLAG: --iptables-min-sync-period="1s"
I0310 20:55:16.410365   84729 flags.go:64] FLAG: --iptables-sync-period="30s"
I0310 20:55:16.410372   84729 flags.go:64] FLAG: --ipvs-exclude-cidrs="[]"
I0310 20:55:16.410387   84729 flags.go:64] FLAG: --ipvs-min-sync-period="0s"
I0310 20:55:16.410394   84729 flags.go:64] FLAG: --ipvs-scheduler=""
I0310 20:55:16.410401   84729 flags.go:64] FLAG: --ipvs-strict-arp="false"
I0310 20:55:16.410409   84729 flags.go:64] FLAG: --ipvs-sync-period="30s"
I0310 20:55:16.410638   84729 flags.go:64] FLAG: --ipvs-tcp-timeout="0s"
I0310 20:55:16.410660   84729 flags.go:64] FLAG: --ipvs-tcpfin-timeout="0s"
I0310 20:55:16.410668   84729 flags.go:64] FLAG: --ipvs-udp-timeout="0s"
I0310 20:55:16.410675   84729 flags.go:64] FLAG: --kube-api-burst="10"
I0310 20:55:16.410683   84729 flags.go:64] FLAG: --kube-api-content-type="application/vnd.kubernetes.protobuf"
I0310 20:55:16.410693   84729 flags.go:64] FLAG: --kube-api-qps="5"
I0310 20:55:16.410718   84729 flags.go:64] FLAG: --kubeconfig=""
I0310 20:55:16.410740   84729 flags.go:64] FLAG: --log-backtrace-at=":0"
I0310 20:55:16.410786   84729 flags.go:64] FLAG: --log-dir=""
I0310 20:55:16.410793   84729 flags.go:64] FLAG: --log-file=""
I0310 20:55:16.410797   84729 flags.go:64] FLAG: --log-file-max-size="1800"
I0310 20:55:16.410802   84729 flags.go:64] FLAG: --log-flush-frequency="5s"
I0310 20:55:16.410807   84729 flags.go:64] FLAG: --logtostderr="true"
I0310 20:55:16.410818   84729 flags.go:64] FLAG: --machine-id-file="/etc/machine-id,/var/lib/dbus/machine-id"
I0310 20:55:16.410963   84729 flags.go:64] FLAG: --masquerade-all="false"
I0310 20:55:16.410981   84729 flags.go:64] FLAG: --master=""
I0310 20:55:16.410986   84729 flags.go:64] FLAG: --metrics-bind-address="127.0.0.1:10249"
I0310 20:55:16.410996   84729 flags.go:64] FLAG: --metrics-port="10249"
I0310 20:55:16.411000   84729 flags.go:64] FLAG: --nodeport-addresses="[]"
I0310 20:55:16.411010   84729 flags.go:64] FLAG: --nodeports-on-localhost="false"
I0310 20:55:16.411014   84729 flags.go:64] FLAG: --one-output="false"
I0310 20:55:16.411018   84729 flags.go:64] FLAG: --oom-score-adj="-999"
I0310 20:55:16.411022   84729 flags.go:64] FLAG: --profiling="false"
I0310 20:55:16.411026   84729 flags.go:64] FLAG: --proxy-mode=""
I0310 20:55:16.411032   84729 flags.go:64] FLAG: --proxy-port-range=""
I0310 20:55:16.411038   84729 flags.go:64] FLAG: --show-hidden-metrics-for-version=""
I0310 20:55:16.411041   84729 flags.go:64] FLAG: --skip-headers="false"
I0310 20:55:16.411256   84729 flags.go:64] FLAG: --skip-log-headers="false"
I0310 20:55:16.411268   84729 flags.go:64] FLAG: --stderrthreshold="2"
I0310 20:55:16.411273   84729 flags.go:64] FLAG: --udp-timeout="250ms"
I0310 20:55:16.411277   84729 flags.go:64] FLAG: --v="1"
I0310 20:55:16.411282   84729 flags.go:64] FLAG: --version="false"
I0310 20:55:16.411291   84729 flags.go:64] FLAG: --vmodule=""
I0310 20:55:16.411295   84729 flags.go:64] FLAG: --write-config-to=""
I0310 20:55:16.411318   84729 server.go:228] "Warning, all flags other than --config, --write-config-to, and --cleanup are deprecated, please begin using a config file ASAP"
I0310 20:55:16.411393   84729 feature_gate.go:245] feature gates: &{map[]}
I0310 20:55:16.411485   84729 feature_gate.go:245] feature gates: &{map[]}
I0310 20:55:16.411599   84729 server.go:563] "Neither kubeconfig file nor master URL was specified, falling back to in-cluster config"

```